web/server/models.py

```diff
@@ -362,3 +372,138 @@ class ArtifactChange(BaseModel):
     path: str
     type: t.Optional[ArtifactType] = None
     file: t.Optional[File] = None
+
+
+class ReportBackfillProgress(BaseModel):
+    completed: int
+    total: int
+    view_name: str
+    start: TimeLike
+    end: t.Optional[TimeLike] = None
+
+
+class ReportTestsRusult(BaseModel):
+    message: str
+
+
+class ReportTestDetails(ReportTestsRusult):
+    details: str
+
+
+class ReportTestsFailure(ReportTestsRusult):
+    total: int
+    failures: int
+    errors: int
+    successful: int
+    dialect: str
+    details: t.List[ReportTestDetails]
+    traceback: str
+
+
+class ReportMeta(BaseModel):
+    status: t.Optional[ReportStatus] = None
+    start_at: t.Optional[int] = None
+    stop_at: t.Optional[int] = None
+    duration: t.Optional[int] = None
+    done: bool = False
+
+    def __init__(self, *arg: t.Any, **kargs: t.Any) -> None:
+        super().__init__(*arg, **kargs)
+        self.status = ReportStatus.init
+        self.start_at = now_timestamp()
+
+
+class Stage(BaseModel):
+    meta: ReportMeta = ReportMeta()
```

Comment:
Not sure if this will work for classes.
```suggestion
    meta: ReportMeta = Field(default_factory=ReportMeta)
```